### PR TITLE
feat(vm/create): Create zvol's instead of the UI

### PR DESCRIPTION
**Pull request by skarekrow**
Beginning of transition from a bunch of private API calls to fewer public ones handling those private parts.

TIcket: #35737

### CODE REVIEW

**Comment by skarekrow** [comment]
It was to reduce number of PR's since it's a harmless quoting change. We
can break it out of this PR if it's a problem.

**Comment by amotin** [comment]
I would not call it a problem, but it is a bad style.  Due to increased patch size it may be difficult to notice some significant code changes.  It requires reviewers to be more attentive then necessary.

**Comment by skarekrow** [comment]
Noted for the future :) Thanks mav!

File: src/middlewared/middlewared/plugins/vm.py [hunk 1]
```diff
@@ -887,20 +887,51 @@ def random_mac(self):
...
         Int('memory'),
         Str('bootloader'),
-        List("devices"),
+        List('devices'),
         Str('vm_type'),
         Bool('autostart'),
+        Bool('create_zvol', required=False),
+        Str('zvol_name', required=False),
+        Str('zvol_type', required=False),
+        Str('zvol_volsize', required=False),
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/vm.py]: That seems counter-intuitive.

Why are the zvol options in the VM schema? I believe those would make more sense if they are part of the `devices` of type `DISK`?

That would also allow one to create multiple disks (zvols).

Comment (skarekrow) [reply src/middlewared/middlewared/plugins/vm.py]: Fair point :)

**Pull request merged**